pkg/cluster/deploystorage_resources.go

```diff
@@ -77,6 +77,10 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 // Legacy storage accounts (public) are not encrypted and cannot be retrofitted.
 // The flag controls this behavior in update/create.
 func (m *manager) storageAccount(name, region string, encrypted bool) *arm.Resource {
+        if(len(m.doc.OpenShiftCluster.Properties.WorkerProfiles) == 0){
+              m.log.Warn("Worker Profiles missing.")
```

Comment:
It would be good to include the fact that we are `Unable to create storage resource due to missing worker profiles.`, so that the log message has some context, and log it as an error not a warning. That said, while it is good practice to defend against calling this function without at least 1 worker profile, none of the callers have any logic handling a `nil` response:
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ce
b1d8c2f0166534512cebee/pkg/cluster/deploystorage.go#L121
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/cluster/deploystorage.go#L124
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/cluster/storageaccounts.go#L84
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/cluster/storageaccounts.go#L85
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/deploy/generator/resources_gateway.go#L829
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/deploy/generator/resources_rp.go#L1866
https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/deploy/generator/resources_rp.go#L1899

I would say that changing the functionality of all of the functions using this one is outside the scope of this PR. The intention is to skip `migrateStorageAccounts` during an AdminUpdate, so it would be prudent to check for the worker profile length [here in that function](https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/cluster/storageaccounts.go#L76) and gracefully skip running it. Like is being done here: https://github.com/Azure/ARO-RP/blob/ac78ec518b52f305ceb1d8c2f0166534512cebee/pkg/cluster/arooperator.go#L18-L21

Reply:
@darthhexx , made the changes, could you please review it and let me know the comments/suggestions